src/main/java/org/opentripplanner/apis/gtfs/mapping/RouteRequestMapper.java

```diff
@@ -283,132 +110,616 @@ private static Stream<String> parseOperation(Collection<Map<String, Collection<S
+        var cost = destinationPolicy.getGraphQLKeepingCost();
+        if (cost != null) {
+          preferences.withArrivingInRentalVehicleAtDestinationCost(cost.toSeconds());
         }
       }
     }
+  }
 
-    private static <T> void call(Map<String, T> m, String name, Consumer<T> consumer) {
-      if (!name.contains(".")) {
-        if (hasArgument(m, name)) {
-          T v = m.get(name);
-          consumer.accept(v);
-        }
-      } else {
-        String[] parts = name.split("\\.");
-        if (hasArgument(m, parts[0])) {
-          Map<String, T> nm = (Map<String, T>) m.get(parts[0]);
-          call(nm, String.join(".", Arrays.copyOfRange(parts, 1, parts.length)), consumer);
+  private static void setScooterOptimization(
+    ScooterPreferences.Builder preferences,
+    GraphQLTypes.GraphQLScooterOptimizationInput args
+  ) {
+    if (args != null) {
+      var type = args.getGraphQLType();
+      var mappedType = type != null ? VehicleOptimizationTypeMapper.map(type) : null;
+      if (mappedType != null) {
+        preferences.withOptimizeType(mappedType);
+      }
+      var triangleArgs = args.getGraphQLTriangle();
+      if (isScooterTriangleSet(triangleArgs)) {
+        preferences.withForcedOptimizeTriangle(triangle -> {
+          triangle
+            .withSlope(triangleArgs.getGraphQLFlatness())
+            .withSafety(triangleArgs.getGraphQLSafety())
+            .withTime(triangleArgs.getGraphQLTime());
+        });
+      }
+    }
+  }
+
+  private static boolean isScooterTriangleSet(
+    GraphQLTypes.GraphQLTriangleScooterFactorsInput args
+  ) {
+    return (
+      args != null &&
+      args.getGraphQLFlatness() != null &&
+      args.getGraphQLSafety() != null &&
+      args.getGraphQLTime() != null
+    );
+  }
+
+  private static void setWalkPreferences(
+    WalkPreferences.Builder preferences,
+    GraphQLTypes.GraphQLWalkPreferencesInput args
+  ) {
+    if (args != null) {
+      var speed = args.getGraphQLSpeed();
+      if (speed != null) {
+        preferences.withSpeed(speed);
+      }
+      var reluctance = args.getGraphQLReluctance();
+      if (reluctance != null) {
+        preferences.withReluctance(reluctance);
+      }
+      var walkSafetyFactor = args.getGraphQLWalkSafetyFactor();
+      if (walkSafetyFactor != null) {
+        preferences.withSafetyFactor(walkSafetyFactor);
+      }
+      var boardCost = args.getGraphQLBoardCost();
+      if (boardCost != null) {
+        preferences.withBoardCost(boardCost.toSeconds());
+      }
+    }
+  }
+
+  private static void setAccessibilityPreferences(
+    RouteRequest request,
+    GraphQLTypes.GraphQLAccessibilityPreferencesInput preferenceArgs
+  ) {
+    if (preferenceArgs != null && preferenceArgs.getGraphQLWheelchair() != null) {
+      request.setWheelchair(preferenceArgs.getGraphQLWheelchair().getGraphQLEnabled());
+    }
+  }
+
+  /**
+   * TODO this doesn't support multiple street modes yet
```

Comment:
Can you throw an exception when there are multiple?

Reply:
Not sure if we want that. For example, we could define `[BICYCLE_RENTAL, WALK]` already now and it would work now and after we add support. However, if you have the modes in different order, it wouldn't use bike rental now. We can discuss this in a dev meeting or here.

Reply:
If you use BICYCLE_RENTAL then WALK is automatically used as a fallback and [BICYCLE_RENTAL,WALK] is the only combination that we support right now.

Reply:
Yep, therefore if I would now throw an exception if more modes are defined, client code would need to be changed if we filter out non bike rental legs if just BICYCLE_RENTAL is defined. However, as @t2gran pointed out in the dev meeting, we can do some "breaking" changes while experimenting.